api/app/src/mappings/garmin/index.ts

```diff
@@ -23,18 +23,18 @@ export interface UserData<T extends MetriportData> {
 }
 
 export const garminUnits = {
-  date: z.string().optional().nullable(), // or regex('yyyy-mm-dd')
+  date: z.string(), // or regex('yyyy-mm-dd')
   time: z.number().int(),
   timeKey: z.string().transform((v) => Number(v)),
-  sleep: z.number().nullable().optional(),
-  kilocalories: z.number().nullable().optional(),
-  duration: z.number().int().nullable().optional(),
-  distance: z.number().int().nullable().optional(),
-  intensityDuration: z.number().int().nullable().optional(),
-  floorsClimbed: z.number().int().nullable().optional(),
-  heartRate: z.number().int().nullable().optional(),
-  steps: z.number().int().nullable().optional(),
-  stressLevel: z.number().int().nullable().optional(),
+  sleep: z.number(),
+  kilocalories: z.number(),
+  duration: z.number().int(),
+  distance: z.number().int(),
```

Comment:
OOC - why the distinction between `number`/`int` here?

Reply:
Good catch, we should allow for partial units on distance!